tritium-registry/src/main/java/com/palantir/tritium/metrics/registry/LongExemplar.java

```diff
@@ -0,0 +1,23 @@
+/*
+ * (c) Copyright 2024 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.tritium.metrics.registry;
+
+public interface LongExemplar<T> {
+    T metadata();
```

Comment:
Let's add log-safety annotations on the `T` generic type, requiring all exemplar data to be logsafe. It's much easier to start with that requirement and loosen it than it is to go in the other direction!